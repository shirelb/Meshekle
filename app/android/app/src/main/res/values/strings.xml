<resources>
    <string name="app_name">משקלה</string>
    <string name="default_notification_channel_id" translatable="false">fcm_default_channel</string>
</resources>
