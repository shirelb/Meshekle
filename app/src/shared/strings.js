const loginScreenStrings = {
    LOGIN: "התחבר",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainScreenStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
};

const drawerMenu = {
    LOGOUT: "התנתק",
    MAIN_SCREEN_NAME: "בית",
    APPOINTMENTS_SCREEN_NAME: "תורים",
    CHORES_SCREEN_NAME: "תורנויות",
    PhoneBook_SCREEN_NAME: "ספר טלפונים",
};

export default {loginScreenStrings,mainScreenStrings,drawerMenu};