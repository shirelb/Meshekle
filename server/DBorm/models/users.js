module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};