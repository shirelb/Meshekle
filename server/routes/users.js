var express = require('express');
var router = express.Router();
var jwt = require('jsonwebtoken');
const Sequelize = require('sequelize');
const Op = Sequelize.Op;
const {sequelize, Users, AppointmentRequests, AppointmentDetails, ScheduledAppointments, Incidents, UsersChoresTypes, Events} = require('../DBorm/DBorm');
var validations = require('./shared/validations');
var constants = require('./shared/constants');
var authentications = require('./shared/authentications');
var helpers = require('./shared/helpers');

/* POST login authenticate a user . */
router.post('/login/authenticate', function (req, res) {
    if (!req.body.userId || !req.body.password) {
        res.status(400).send({"message": constants.usersRoute.AUTHENTICATION_FAILED});
    } else {
        validations.checkIfUserExist(req.body.userId, res)
            .then(user => {
                if (user.dataValues) {
                    var payload = {
                        userId: user.userId,
                        userFullname: user.fullname,
                    };
                    user.userId === req.body.userId &&
                    user.password === req.body.password ?
                        authentications.sendToken(payload, res)
                        :
                        res.status(400).send({
                            "success": false,
                            "message": constants.usersRoute.AUTHENTICATION_FAILED,
                            "token": 'null'
                        });

                }
            });
    }
});

router.use(function (req, res, next) {
    authentications.verifyToken(req, res, next);
});

router.post('/validToken', function (req, res) {
    res.status(200).send({
        message: constants.usersRoute.VALID_TOKEN,
        payload: req.decoded.payload
    });
});

/* GET users listing. */
router.get('/', function (req, res, next) {
    Users.findAll()
        .then(users => {
            console.log(users);
            res.status(200).send(users);
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/* GET users by name listing. */
router.get('/name/:name', function (req, res, next) {
    Users.findAll({
        where: {
            fullname: {
                [Op.like]: '%' + req.params.name + '%',
            },
        }
    })
        .then(users => {
            console.log(users);
            res.status(200).send(users);
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/* GET users by name listing. */
router.get('/userId/:userId', function (req, res, next) {
    Users.findAll({
        where: {
            userId: req.params.userId,
        }
    })
        .then(users => {
            console.log(users);
            res.status(200).send(users);
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

// update user by userId.
router.put('/update/userId/:userId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            let updateFields = {};
            req.body.fullname ? updateFields.fullname = req.body.fullname : null;
            req.body.email ? updateFields.email = req.body.email : null;
            req.body.mailbox ? updateFields.mailbox = req.body.mailbox : null;
            req.body.cellphone ? updateFields.cellphone = req.body.cellphone : null;
            req.body.phone ? updateFields.phone = req.body.phone : null;
            req.body.bornDate ? updateFields.bornDate = req.body.bornDate : null;
            req.body.active ? updateFields.active = req.body.active : null;

            Users.update(
                updateFields,
                {
                    where: {
                        userId: req.params.userId
                    }
                })
                .then(isUpdated => {
                    res.status(200).send({
                        "message": constants.usersRoute.USER_UPDATE_SUCCESS,
                        "result": isUpdated[0]
                    });
                })
                .catch(err => {
                    console.log(err);
                    res.status(500).send(err);
                })
        });
});


/* POST appointment request of user . */
router.post('/appointments/request', function (req, res, next) {
    createAppointmentRequestId()
        .then(appointmentRequestId => {
            helpers.createAppointmentDetails(appointmentRequestId, req, res)
                .then(newAppointmentDetails => {
                    if (newAppointmentDetails) {
                        AppointmentRequests.create({
                            requestId: appointmentRequestId,
                            optionalTimes: JSON.stringify(req.body.availableTime),
                            notes: req.body.notes,
                            status: "requested",
                        })
                            .then((newAppointmentRequest) => {
                                res.status(200).send({
                                    "message": constants.usersRoute.SUCCESSFUL_APPOINTMENT_REQUEST,
                                    newAppointmentDetails, newAppointmentRequest
                                });
                            })
                            .catch(err => {
                                console.log(err);
                                res.status(500).send(err);
                            })
                    }
                })
        })
});

/* POST appointment set of user . */
router.post('/appointments/set', function (req, res, next) {
    helpers.createAppointmentSetId()
        .then(appointmentSetId => {
            helpers.createAppointmentDetails(appointmentSetId, req, res)
                .then(newAppointmentDetails => {
                    if (newAppointmentDetails) {
                        ScheduledAppointments.create({
                            appointmentId: appointmentSetId,
                            startDateAndTime: new Date(req.body.date + "T" + req.body.startHour),
                            endDateAndTime: new Date(req.body.date + "T" + req.body.endHour),
                            remarks: req.body.notes,
                            status: "set",
                        })
                            .then((newAppointment) => {
                                Events.create({
                                    userId: req.body.userId,
                                    eventType: "Appointments",
                                    eventId: newAppointment.appointmentId
                                })
                                    .then((newEvent) => {
                                        res.status(200).send({
                                            "message": constants.usersRoute.SUCCESSFUL_APPOINTMENT,
                                            newAppointmentDetails,
                                            newAppointment,
                                            newEvent
                                        });
                                    })
                                    .catch(err => {
                                        console.log(err);
                                        res.status(500).send(err);
                                    });
                            })
                            .catch(err => {
                                console.log(err);
                                res.status(500).send(err);
                            })
                    }
                })
        })
});

/* POST incident of user . */
router.post('/incidents/open', function (req, res, next) {
    Incidents.create({
        userId: req.body.userId,
        category: req.body.category,
        description: req.body.description,
        status: "opened",
    })
        .then((newIncident) => {
            console.log('New incident ' + newIncident.incidentId + ', of userId' + newIncident.userId + ' has been created.');
            Events.create({
                userId: newIncident.userId,
                eventType: "Incidents",
                eventId: newIncident.incidentId
            })
                .then((newEvent) => {
                    res.status(200).send({"message": constants.usersRoute.SUCCESSFUL_INCIDENT, newIncident, newEvent});
                })
                .catch(err => {
                    console.log(err);
                    res.status(500).send(err);
                });
        })
        .catch(err => {
            res.status(400).send({
                "message": constants.usersRoute.USER_NOT_FOUND,
                err
            });
        })
});

/* POST appointment user approve . */
router.post('/appointments/approve', function (req, res, next) {
    AppointmentRequests.findOne({
        where: {
            requestId: req.body.appointmentRequestId,
        },
        include: [{
            model: AppointmentDetails,
            where: {
                clientId: req.body.userId,
                appointmentId: req.body.appointmentRequestId,
            },
            required: true,
        }]
    })
        .then(appointmentsRequest => {
            appointmentsRequest.update({
                status: "approved"
            });

            ScheduledAppointments.create({
                appointmentId: appointmentsRequest.requestId,
                startDateAndTime: new Date(req.body.date + "T" + req.body.startHour),
                endDateAndTime: new Date(req.body.date + "T" + req.body.endHour),
                remarks: appointmentsRequest.notes,
                status: "set",
            })
                .then((newAppointment) => {
                    Events.create({
                        userId: appointmentsRequest.AppointmentDetail.clientId,
                        eventType: "Appointments",
                        eventId: newAppointment.appointmentId
                    })
                        .then((neEvent) => {
                            res.status(200).send({
                                "message": constants.usersRoute.SUCCESSFUL_APPOINTMENT,
                                appointmentsRequest,
                                newAppointment,
                                neEvent
                            });
                        })
                        .catch(err => {
                            console.log(err);
                            res.status(500).send(err);
                        });

                })
                .catch(err => {
                    console.log(err);
                    res.status(500).send(err);
                });
        })
        .catch(err => {
            console.log(err);
            res.status(400).send({
                "message": constants.usersRoute.APPOINTMENT_REQUEST_NOT_FOUND,
                err
            });
        });
});

/* POST appointment user reject . */
router.post('/appointments/reject', function (req, res, next) {
    AppointmentRequests.findOne({
        where: {
            requestId: req.body.appointmentRequestId,
        },
        include: [{
            model: AppointmentDetails,
            where: {
                clientId: req.body.userId,
                appointmentId: req.body.appointmentRequestId,
            },
            required: true
        }]
    })
        .then(appointmentsRequest => {
            console.log(appointmentsRequest);
            appointmentsRequest.update({
                status: "rejected"
            });
            res.status(200).send({
                "message": constants.usersRoute.SUCCESSFUL_REJECT_APPOINTMENT_REQUEST,
                appointmentsRequest
            });
        })
        .catch(err => {
            console.log(err);
            res.status(400).send({
                "message": constants.usersRoute.APPOINTMENT_REQUEST_NOT_FOUND,
                err
            });
        });
});

/* GET appointments of user by userId listing. */
router.get('/appointments/userId/:userId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            if (user.dataValues) {
                let whereClause = {};
                req.query.status ? whereClause.status = req.query.status : null;
                req.query.appointmentId ? whereClause.appointmentId = req.query.appointmentId : null;
                ScheduledAppointments.findAll({
                    where: whereClause,
                    include: [
                        {
                            model: AppointmentDetails,
                            where: {
                                clientId: req.params.userId,
                            },
                            required: true
                        }
                    ]
                })
                    .then(userAppointments => {
                        console.log(userAppointments);
                        res.status(200).send(userAppointments);
                    })
                    .catch(err => {
                        res.status(500).send(err);
                    })
            }
        })
});

/* GET incidents of user by userId listing. */
router.get('/incidents/userId/:userId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            if (user.dataValues) {
                let whereClause = {};
                whereClause.userId = req.params.userId;
                req.query.status ? whereClause.status = req.query.status : null;
                req.query.incidentId ? whereClause.incidentId = req.query.incidentId : null;
                Incidents.findAll({
                    where: whereClause
                })
                    .then(userIncidents => {
                        console.log(userIncidents);
                        res.status(200).send(userIncidents);
                    })
                    .catch(err => {
                        console.log(err);
                        res.status(500).send(err);
                    })
            }
        });
});

function createAppointmentRequestId() {
    return AppointmentRequests.max('requestId')
        .then(max => {
            return isNaN(max) ?
                1
                :
                max + 2
        })
        .catch(err => {
            console.log(err);
        })
}

/* PUT cancel appointment of user by userId listing. */
router.put('/appointments/cancel/userId/:userId/appointmentId/:appointmentId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            if (user.dataValues) {
                ScheduledAppointments.findOne({
                    where: {
                        appointmentId: req.params.appointmentId,
                    },
                    include: {
                        model: AppointmentDetails,
                        where: {
                            clientId: req.params.userId,
                            appointmentId: req.params.appointmentId,
                        },
                        required: true
                    }
                })
                    .then(appointment => {
                        if (appointment) {
                            if (appointment.status === "set") {
                                appointment.update({
                                    status: "canceled"
                                });
                                Events.destroy({
                                    where: {
                                        userId: req.params.userId,
                                        eventType: "Appointments",
                                        eventId: appointment.appointmentId
                                    }
                                })
                                    .then((newEvent) => {
                                        res.status(200).send({
                                            "message": constants.usersRoute.SUCCESSFUL_CANCEL_APPOINTMENT,
                                            appointment
                                        });
                                    })
                                    .catch(err => {
                                        console.log(err);
                                        res.status(500).send(err);
                                    });
                            } else {
                                appointment.status === "canceled" ?
                                    res.status(400).send({
                                        "message": constants.usersRoute.ALREADY_CANCELED_APPOINTMENT,
                                        appointment
                                    })
                                    :
                                    res.status(400).send({
                                        "message": constants.usersRoute.PASSED_APPOINTMENT,
                                        appointment
                                    });
                            }
                        } else {
                            res.status(400).send({
                                "message": constants.usersRoute.APPOINTMENT_NOT_FOUND,
                            });
                        }
                    })
                    .catch(err => {
                        console.log(err);
                        res.status(400).send({
                            "message": constants.usersRoute.APPOINTMENT_NOT_FOUND,
                            err
                        });
                    })
            }
        })
});

/* PUT cancel incident of user by userId listing. */
router.put('/incidents/cancel/userId/:userId/incidentId/:incidentId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            if (user.dataValues) {
                Incidents.findOne({
                    where: {
                        userId: req.params.userId,
                        incidentId: req.params.incidentId,
                    }
                })
                    .then(incident => {
                        if (incident) {
                            if (incident.status === "opened") {
                                incident.update({
                                    status: "canceled"
                                });

                                Events.destroy({
                                    where: {
                                        userId: req.params.userId,
                                        eventType: "Incidents",
                                        eventId: incident.incidentId
                                    }
                                })
                                    .then((newEvent) => {
                                        res.status(200).send({
                                            "message": constants.usersRoute.SUCCESSFUL_CANCEL_INCIDENT,
                                            incident
                                        });
                                    })
                                    .catch(err => {
                                        console.log(err);
                                        res.status(500).send(err);
                                    });
                            } else {
                                incident.status === "canceled" ?
                                    res.status(400).send({
                                        "message": constants.usersRoute.ALREADY_CANCELED_INCIDENT,
                                        incident
                                    })
                                    :
                                    res.status(400).send({
                                        "message": constants.usersRoute.ALREADY_RESOLVED_INCIDENT,
                                        incident
                                    });
                            }
                        } else {
                            // if (user !== null)
                            res.status(400).send({
                                "message": constants.usersRoute.INCIDENT_NOT_FOUND,
                            });
                        }
                    })
                    .catch(err => {
                        console.log(err);
                        res.status(400).send({
                            "message": constants.usersRoute.INCIDENT_NOT_FOUND,
                            err
                        });
                    })
            }
        })
});

/*
/!* GET releases of user by userId and choreTypeId listing. *!/
router.get('/userId/:userId/chores/choreTypeId/:choreTypeId/releases', function (req, res, next) {
    UsersReleases.findAll({
        where: {
            userId: req.params.userId,
            choreTypeId: req.params.choreTypeId,
        }
    })
        .then(userReleases => {
            res.status(200).send(userReleases);
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/!* PUT releases of user by userId and choreTypeId listing. *!/
router.put('/userId/:userId/chores/choreTypeId/:choreTypeId/releases', function (req, res, next) {
    UsersReleases.findAll({
        where: {
            userId: req.params.userId,
            choreTypeId: req.params.choreTypeId,
        }
    })
        .then(userReleases => {
            userReleases ?
                userReleases.update({
                    startDate: req.body.startDate,
                    endDate: req.body.endDate
                })
                    .then(update => {
                        res.status(200).send({"message": "releases of user updated successfully!", update});
                    })
                    .catch(err => {
                        console.log(err);
                        res.status(500).send(err);
                    })
                :
                UsersReleases.create({
                    userId: req.params.userId,
                    choreTypeId: req.params.choreTypeId,
                    startDate: req.body.startDate,
                    endDate: req.body.endDate
                })
                    .then(newUserRelease => {
                        res.status(200).send({"message": "releases of user added successfully!", newUserRelease});
                    })
                    .catch(err => {
                        console.log(err);
                        res.status(500).send(err);
                    })
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/!* GET user chores by userId listing. *!/
router.get('/userId/:userId/chores', function (req, res, next) {
    UsersChores.findAll({
        where: {
            userId: req.params.userId,
        }
    })
        .then(userChores => {
            res.status(200).send(userChores);
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/!* GET user choresTypes by userId listing. *!/
router.get('/userId/:userId/choresTypes', function (req, res, next) {
    UsersChores.findAll({
        where: {
            userId: req.params.userId,
        }
    })
        .then(userChoresTypes => {
            while (i < req.body.questions.length) {
                arr.push(DButilsAzure.execQuery("" +
                    "INSERT INTO QaRestorePassword (userId,question,answer)" +
                    " VALUES (" + user_id + ",'" + req.body.questions[i] + "','" + req.body.answers[i] + "')"));
                i++;
            }
            for (choreTypeId in userChoresTypes) {
                ChoreTypes.findAll({
                    where: {
                        userId: req.params.userId,
                    }
                })
            }

            res.status(200).send(userChores);
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/!* PUT cancel incident of user by userId listing. *!/
router.put('/users/userId/:userId/update', function (req, res, next) {
    Users.findOne({
        where: {
            userId: req.params.userId,
        }
    }).then(user => {
        user.update({
            password: req.body.password ? req.body.password : user.password,
            email: req.body.email ? req.body.email : user.email,
            mailbox: req.body.mailbox ? req.body.mailbox : user.mailbox,
            cellphone: req.body.cellphone ? req.body.cellphone : user.cellphone,
            phone: req.body.phone ? req.body.phone : user.phone,
        });
        res.status(200).send({"message": "Incident canceled successfully!", appointment});
    })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});
*/

/* GET events of user by userId listing. */
router.get('/events/userId/:userId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            if (user.dataValues) {
                Events.findAll({
                    where: {
                        userId: req.params.userId,
                    },
                })
                    .then(userEvents => {
                        console.log(userEvents);
                        res.status(200).send(userEvents);
                    })
                    .catch(err => {
                        res.status(500).send({
                            "message": constants.general.SOMETHING_WENT_WRONG,
                            err
                        });
                    })
            }
        })
});

/* GET all appointment requests of user . */
router.get('/appointmentRequests/userId/:userId', function (req, res, next) {
    validations.checkIfUserExist(req.params.userId, res)
        .then(user => {
            if (user.dataValues) {
                let whereClause = {};
                req.query.status ? whereClause.status = req.query.status : null;
                req.query.appointmentRequestId ? whereClause.requestId = req.query.appointmentRequestId : null;
                AppointmentRequests.findAll({
                    where: whereClause,
                    include: [
                        {
                            model: AppointmentDetails,
                            where: {
                                clientId: req.params.userId,
                            },
                            required: true
                        }
                    ]
                })
                    .then(userAppointments => {
                        console.log(userAppointments);
                        res.status(200).send(userAppointments);
                    })
                    .catch(err => {
                        res.status(500).send(err);
                    })
            }
        })
});


module.exports = router;
