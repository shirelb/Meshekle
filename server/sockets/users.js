
module.exports = function (socket,pushToAllWebClient,pushToAllAppClient) {
    socket.on('userShallowDeleted', (data) => {
        console.log('socket userShallowDeleted data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });

    socket.on('userCreated', (data) => {
        console.log('socket userCreated data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });

    socket.on('userUpdated', (data) => {
        console.log('socket userUpdated data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });


};
