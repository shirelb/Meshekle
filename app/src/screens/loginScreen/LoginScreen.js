import React, {Component} from 'react';
import {Button, ScrollView, Text, TextInput, View} from 'react-native';
import axios from 'axios';
import styles from './LoginCSS';
import {SERVER_URL} from '../../shared/constants'

export default class LoginScreen extends Component {
    /*componentDidMount(){
        const headers = {
            'Authorization': 'Bearer ' + this.props.jwt
        };
    }*/

    onSubmitPress = () => {
        console.log("in onSubmitPress");
        return axios.get(`${SERVER_URL}/api/users/login/authenticate`,
            {
                "userId": "436547125",
                "password": "tset22"
            }
        )
            .then((response) => {
                console.log("after get response");
                console.log(response);
                // deviceStorage.saveKey("id_token", response.data.jwt);
                this.props.onLoginPress();
            })
            .catch((error) => {
                console.log("after get error");
                console.log(error);
            });
    };


    render() {
        return (
            <ScrollView style={{padding: 20}}>
                <Text style={styles.loginLabel}>
                    Login
                </Text>
                <TextInput  placeholder='Username'/>
                <TextInput placeholder='Password'/>
                <View style={{margin: 7}}/>
                <Button
                    onPress={this.onSubmitPress}
                    title="Submit"
                />
            </ScrollView>
        )
    }
}
