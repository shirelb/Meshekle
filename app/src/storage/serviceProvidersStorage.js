import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getServiceProviders = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('load appointments error ', error)
        });
};


export default {getServiceProviders};
