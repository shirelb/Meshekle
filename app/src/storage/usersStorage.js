import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
