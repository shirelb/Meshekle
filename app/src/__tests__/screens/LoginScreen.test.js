import React from 'react';

import {shallow} from "enzyme/build";
import LoginScreen from "../../screens/loginScreen/LoginScreen";
import strings from "../../shared/strings";
import serviceProvidersStorage from "../../storage/serviceProvidersStorage";
import usersStorage from "../../storage/usersStorage";
import phoneStorage from "react-native-simple-store";
import users from "../jsons/users";

jest.mock("../../storage/serviceProvidersStorage");
jest.mock("../../storage/usersStorage");
jest.mock("react-native-simple-store");


describe("LoginScreen should", () => {
    let wrapper = null;
    let componentInstance = null;
    const props = {};
    const userIdTest = "549963652";
    const passwordTest = "Qw345678";

    const navigation = {navigate: jest.fn()};
    usersStorage.getUsers.mockResolvedValue(users);
    phoneStorage.get.mockImplementation((key) => Promise.resolve(mockStore[key]));
    phoneStorage.update.mockImplementation((key, value) => Promise.resolve(mockStore[key] = value));

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('match snapshot', async () => {
        wrapper = shallow(<LoginScreen/>);
        expect(wrapper).toMatchSnapshot();
    });

    test("render what the user see", async () => {
        wrapper = shallow(<LoginScreen/>);
        componentInstance = wrapper.instance();

        expect(wrapper.find('Image')).toHaveLength(1);
        expect(wrapper.find('Text')).toHaveLength(1);
        expect(wrapper.find('Text').props().children).toEqual(strings.loginScreenStrings.APP_NAME);
        expect(wrapper.find('FormTextInput')).toHaveLength(2);
        expect(wrapper.find('FormTextInput').at(0).props().placeholder).toEqual(strings.loginScreenStrings.USER_ID_PLACEHOLDER);
        expect(wrapper.find('FormTextInput').at(0).props().value).toEqual('');
        expect(wrapper.find('FormTextInput').at(1).props().placeholder).toEqual(strings.loginScreenStrings.PASSWORD_PLACEHOLDER);
        expect(wrapper.find('FormTextInput').at(1).props().value).toEqual('');
        expect(wrapper.find('Button')).toHaveLength(2);
        expect(wrapper.find('Button').at(0).props().label).toEqual(strings.loginScreenStrings.LOGIN);
        expect(wrapper.find('Button').at(1).props().label).toEqual(strings.loginScreenStrings.FORGOT_PASSWORD);

        expect(componentInstance.state.err.length).toEqual(0);
        expect(componentInstance.state.err).toEqual([]);
        expect(componentInstance.state.userId).toEqual('');
        expect(componentInstance.state.password).toEqual('');
    });

    test("handle change of userId", async () => {
        wrapper = shallow(<LoginScreen/>);
        componentInstance = wrapper.instance();

        expect(componentInstance.state.userId).toEqual('');

        await wrapper.find('FormTextInput').at(0).props().onChangeText(userIdTest);
        expect(componentInstance.state.userId).toEqual(userIdTest);
    });

    test("handle change of password", async () => {
        wrapper = shallow(<LoginScreen/>);
        componentInstance = wrapper.instance();

        expect(componentInstance.state.password).toEqual('');

        await wrapper.find('FormTextInput').at(1).props().onChangeText(passwordTest);
        expect(componentInstance.state.password).toEqual(passwordTest);
    });

    test("validate userId and password", async () => {
        wrapper = shallow(<LoginScreen/>);
        componentInstance = wrapper.instance();

        let validateResponse = componentInstance.validate("", passwordTest);
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.EMPTY_USER_ID);

        validateResponse = componentInstance.validate("11111111111111111", passwordTest);
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate("user223", passwordTest);
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, "");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.EMPTY_PASSWORD);

        validateResponse = componentInstance.validate(userIdTest, "5");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, "55555555555555555555555555");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, "סיסמא");
        expect(validateResponse.length).toEqual(1);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate("", "סיסמא");
        expect(validateResponse.length).toEqual(2);
        expect(validateResponse[0]).toEqual(strings.loginScreenStrings.EMPTY_USER_ID);
        expect(validateResponse[1]).toEqual(strings.loginScreenStrings.WRONG_CREDENTIALS);

        validateResponse = componentInstance.validate(userIdTest, passwordTest);
        expect(validateResponse.length).toEqual(0);
    });

    test("submit login form with wrong credentials", async () => {
        wrapper = shallow(<LoginScreen/>);
        componentInstance = wrapper.instance();
        const onSubmitPressSpy = jest.spyOn(componentInstance, 'onSubmitPress');

        await componentInstance.setState({userId: "", password: ""});
        const validateSpy = jest.spyOn(componentInstance, 'validate');
        validateSpy.mockReturnValue([strings.loginScreenStrings.EMPTY_USER_ID, strings.loginScreenStrings.EMPTY_PASSWORD]);

        await wrapper.find('Button').at(0).props().onPress();

        expect(onSubmitPressSpy).toHaveBeenCalled();
        expect(componentInstance.state.err).toHaveLength(2);
        expect(wrapper.find('Text')).toHaveLength(3);
        expect(wrapper.find('Text').at(1).props().children).toEqual(strings.loginScreenStrings.EMPTY_USER_ID);
        expect(wrapper.find('Text').at(2).props().children).toEqual(strings.loginScreenStrings.EMPTY_PASSWORD);
    });

    test("submit login form with right credentials", async () => {
        usersStorage.userLogin.mockResolvedValue({
            "success": true,
            "message": "Token generated successfully !",
            "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzZXJ2aWNlUHJvdmlkZXJJZCI6IjU0OTk2MzY1MiIsInVzZXJJZCI6IjU0OTk2MzY1MiIsImlhdCI6MTU1ODYyNjk3MSwiZXhwIjoxNTU4NjYyOTcxfQ.r4oH3N4qvGwRVAFMMRsH5Ls7hC1SQGDme7Gw_bipOc0"
        });
        usersStorage.userValidToken.mockResolvedValue({
            data: {
                payload: {
                    userId: "549963652",
                    userFullname: "Padget Creaser",
                }
            }
        });

        wrapper = shallow(<LoginScreen navigation={navigation}/>);
        componentInstance = wrapper.instance();
        const onSubmitPressSpy = jest.spyOn(componentInstance, 'onSubmitPress');

        await componentInstance.setState({userId: userIdTest, password: passwordTest});
        const validateSpy = jest.spyOn(componentInstance, 'validate');
        validateSpy.mockReturnValue([]);

        await wrapper.find('Button').at(0).simulate('click', {
            preventDefault() {
            }
        });

        expect(onSubmitPressSpy).toHaveBeenCalled();

        expect(componentInstance.state.err.length).toEqual(0);
        expect(componentInstance.state.err).toEqual([]);
        expect(wrapper.find('.errorText')).toHaveLength(0);
    });
});