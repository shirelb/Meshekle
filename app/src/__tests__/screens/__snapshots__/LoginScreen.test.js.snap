// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/logo4000.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "resizeMode": "contain",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#3e3e3e",
        "fontSize": 25,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    משקל'ה
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <FormTextInput
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="ת.ז."
      value=""
    />
    <FormTextInput
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="סיסמא"
      secureTextEntry={true}
      value=""
    />
    <Button
      label="התחבר"
      onPress={[Function]}
    />
    <Button
      label="שכחתי סיסמא"
      onPress={[Function]}
    />
  </View>
</View>
`;
